driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -669,7 +623,7 @@ public long getReadTimeoutMillis() {
         private Bootstrap newBootstrap() {
             Bootstrap b = new Bootstrap();
             b.group(eventLoopGroup)
-                .channel(CHANNEL_CLASS);
+                .channel(channelClass);
```

Comment:
I was doing that initially, then I changed; if you read the docs for NettyOptions, they currently state that channelClass() is called only once when the cluster is created. Which IMHO seems logical from the client point of view. Otherwise we would have to have the docs say: "this method is called for every connection created" - but then users would legitimately argue: why call it for every connection since it always returns the same result?